components/encoder/ams/ams_as5048.go

```diff
@@ -125,7 +125,7 @@ func newAS5048Encoder(
 	conf resource.Config,
 	logger *zap.SugaredLogger,
 ) (encoder.Encoder, error) {
-	cancelCtx, cancel := context.WithCancel(ctx)
+	cancelCtx, cancel := context.WithCancel(context.Background())
 	res := &Encoder{
 		Named:        conf.ResourceName().AsNamed(),
 		cancelCtx:    cancelCtx,
```

Comment:
Should `startPositionLoop` below be using the original `ctx` or `cancelCtx`?

Reply:
Very good catch! `startPositionLoop` does some setup stuff that takes the current context, and `ctx` should still be used there, but then it starts a background coroutine that does other stuff, and _that_ one should use `cancelCtx`. Thanks! I'll update after lunch.

Reply:
...no, now that I look closer, it already is doing stuff right. That's the change down on line 190 (whereas line 181 should use the ephemeral context instead). I already checked this!